bin/kafka-server-start.sh

```diff
@@ -37,6 +37,10 @@ case $COMMAND in
     EXTRA_ARGS="-daemon "$EXTRA_ARGS
     shift
     ;;
+  --soft-file-limit)
```

Comment:
I think this doesn't work if you provide them in the reverse order (i.e., `--soft-file-limit <number> -daemon`, which I would assume would be possible)? I think you'd need a while loop around this.